### PR TITLE
Language Toggle

### PR DESCRIPTION
**Pull request by imacrayon**
Added a language toggle to the main navigation.

### CODE REVIEW

File: app/ScheduledMessage.php [hunk 1]
```diff
@@ -32,8 +34,11 @@ public static function scopeSent($query)
 
     public function getHtmlAttribute()
     {
-        // TODO: show Spanish version here too
-        // LinkFinder in `yarri/link-finder` package
-        return (new \LinkFinder)->process($this->attributes['body_en']);
+        $key = 'body_'.App::getLocale();
+        if (! array_key_exists($key, $this->attributes)) {
+            $key = 'body_en';
+        }
```
Comment (sethetter) [inline app/ScheduledMessage.php]: Nice!

File: config/voteict.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+return [
+
+    'locales' => [
+        'English' => 'en',
+        'Español' => 'es',
+    ],
```
Comment (sethetter) [inline config/voteict.php]: Also nice!

File: app/Message.php [hunk 3]
```diff
@@ -35,9 +35,7 @@ public function subscriber()
      */
     public function getLocaleFromTrigger($key)
     {
-        $locales = array_diff(scandir(resource_path('lang')), ['..', '.']);
-
-        foreach ($locales as $locale) {
+        foreach (config('voteict.locales') as $locale) {
```
Comment (sethetter) [inline app/Message.php]: :tada: So much better

**Comment by sethetter** [comment]
Just needs a merge from master then this is good to go!

**Comment by imacrayon** [comment]
I can't find a way to write tests for this translation stuff though :disappointed: PHPUnit fakes HTTP requests is a weird way where the Request object doesn't get build up until middleware runs, but we have to check the request *before* the middleware so we can dynamically create the routes in `/routes/web.php`. All this works in a normal server environment but not when running tests :man_shrugging:

**Comment by sethetter** [comment]
Do the routes _have_ to be dynamically generated? Or could we just loop through `config.locales` and register them for each?

**Comment by imacrayon** [comment]
They don't have to be dynamically generated but if we use a loop, each route would need a unique name like `archive.en` & `archive.es` then things get messy when you want to link to a page using the route name.

I did find one solution where you could override the `route()` helper to handle the currently set local. Basically when you pass it a route name it would append the locale code to the end of the name before looking loading the route. I was trying to avoid have to hacking core helpers like that but it might be worth it if testing works.

**Comment by sethetter** [comment]
I know it's generally bad practice, but tests can come later too. Small side projects like this deserve a little leniency.

**Comment by sethetter** [comment]
Whenever you're happy with it, feel free to merge!

**Pull request merged**